### PR TITLE
ddl: Update error information and add tests

### PR DESCRIPTION
**Pull request by zimulala**
update error codes and add tests
improve the test coverage from 79% to 81%

### CODE REVIEW

File: ddl/column.go [hunk 1]
```diff
@@ -217,13 +217,13 @@ func (d *ddl) onDropColumn(t *meta.Meta, job *model.Job) error {
 	colInfo := findCol(tblInfo.Columns, colName.L)
 	if colInfo == nil {
 		job.State = model.JobCancelled
-		return infoschema.ErrColumnNotExists.Gen("column %s doesn't exist", colName)
+		return ErrCantDropFieldOrKey.Gen("index %s doesn't exist", colName.O)
```
Comment (shenli) [inline ddl/column.go]: Why change colum to index? This error is cause by drop an nonexistent column.

File: ddl/ddl.go [hunk 2]
```diff
@@ -782,7 +784,7 @@ func (d *ddl) buildTableInfo(tableName model.CIStr, cols []*table.Column, constr
 				key := constr.Keys[0]
 				col := table.FindCol(cols, key.Column.Name.O)
 				if col == nil {
-					return nil, infoschema.ErrColumnNotExists.Gen("no such column: %v", key)
+					return nil, errKeyColumnDoesNotExits.Gen("key column %s doesn't exist in table", key.Column.Name)
```
Comment (shenli) [inline ddl/ddl.go]: Add table name in error message.

File: ddl/index.go [hunk 3]
```diff
@@ -145,7 +144,7 @@ func (d *ddl) onCreateIndex(t *meta.Meta, job *model.Job) error {
 			if idx.State == model.StatePublic {
 				// we already have a index with same index name
 				job.State = model.JobCancelled
-				return infoschema.ErrIndexExists.Gen("CREATE INDEX: index already exist %s", indexName)
+				return errDupKeyName.Gen("index already exist %s", indexName)
```
Comment (shenli) [inline ddl/index.go]: use indexName.O

**Comment by shenli** [comment]
LGTM

**Comment by coocood** [comment]
CIStr types have `String` method, we don't need to get `.O` string for format arg.

**Comment by coocood** [comment]
LGTM

**Pull request merged**